integrations/dell_idrac/dell_idrac_test.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import annotations
+import unittest
+from unittest.mock import MagicMock, patch
+from kdnrm.secret import Secret
+from kdnrm.saas_type import SaasUser
+from kdnrm.exceptions import SaasException
+from dell_idrac import SaasPlugin, DelliDRACClient
+from kdnrm.log import Log
+
+class DellIDRACTest(unittest.TestCase):
+
+    def setUp(self):
+        Log.init()
+        Log.set_log_level("DEBUG")
+
+    @staticmethod
+    def plugin(prior_password: Secret = None):
+        user = SaasUser(
+            username=Secret("testuser"),
+            new_password=Secret("NewPassword123"),
+            prior_password=Secret(("SomeOtherPassword", "OldPassword123")) 
+        )
+        config_record = MagicMock()
+        config_record.get_custom_field_value.side_effect = lambda key, single=True: {
+            "login": "admin",
+            "password": "adminpass",
+            "idrac_ip": "1.2.3.4",
+            "user_id": "42"
+        }.get(key)
+        user.fields = []
+        return SaasPlugin(user=user, config_record=config_record)
+
+    def test_requirements(self):
+        req_list = SaasPlugin.requirements()
+        self.assertEqual(req_list, ["requests"])
+
+    def test_change_password_success(self):
+        plugin = self.plugin()
+        with patch.object(DelliDRACClient, "get_user_id_from_user_fields", return_value="42"), \
+             patch.object(DelliDRACClient, "check_username_by_id"), \
+             patch.object(DelliDRACClient, "change_dell_idrac_user_password") as mock_change_pw:
+            plugin.change_password()
+            mock_change_pw.assert_called_once_with("42", "NewPassword123")
+            self.assertEqual(plugin._SaasPlugin__user_id, "42")
+            self.assertTrue(any(f.label == "user_id" for f in plugin.return_fields))
```

Comment:
The test does not check whether the password was changed to the new password "NewPassword123"